src/main/java/net/javadiscord/javabot/command/moderation/ModerateCommand.java

```diff
@@ -0,0 +1,44 @@
+package net.javadiscord.javabot.command.moderation;
+
+import net.dv8tion.jda.api.entities.ChannelType;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
+import net.dv8tion.jda.api.requests.restaction.interactions.ReplyCallbackAction;
+import net.javadiscord.javabot.command.ResponseException;
+import net.javadiscord.javabot.command.Responses;
+import net.javadiscord.javabot.command.interfaces.ISlashCommand;
+
+/**
+ * Abstract class, that represents a single moderation command.
+ */
+public abstract class ModerateCommand implements ISlashCommand {
+	private boolean allowThreads = true;
+
+	@Override
+	public ReplyCallbackAction handleSlashCommandInteraction(SlashCommandInteractionEvent event) throws ResponseException {
+		if (event.getGuild() == null) {
+			return Responses.error(event, "This command can only be used inside servers.");
+		}
+
+		if (allowThreads) {
+			if (event.getChannelType() != ChannelType.TEXT && event.getChannelType() != ChannelType.GUILD_PRIVATE_THREAD && event.getChannelType() != ChannelType.GUILD_PUBLIC_THREAD) {
+				return Responses.error(event, "This command can only be performed in a server text channel or thread.");
+			}
+		} else {
+			if (event.getChannelType() != ChannelType.TEXT) {
+				return Responses.error(event, "This command can only be performed in a server text channel.");
+			}
+		}
+		Member member = event.getMember();
+		if (member == null) {
+			return Responses.error(event, "Unexpected error has occurred, Slash Command member was null.");
+		}
```

Comment:
Make this error message a bit more user friendly

Reply:
There are 4 comments there, which...

There are as friendly as they can be, and it should never reach `member == null`, unless the event was emitted.